tasks/upload-sbom-to-atlas/upload-sbom-to-atlas.yaml

```diff
@@ -0,0 +1,261 @@
+      description: URL of the SSO token issuer
+      type: string
+    - name: skipUploadingSbom
+      default: "true"
+      description: >-
+        If enabled, all steps of the task are skipped.
+      type: string
+    - name: supportedCycloneDxVersion
+      default: "1.4"
+      description: >-
+        If the SBOM uses a higher CycloneDX version, `syft convert` in the task
+        will convert all SBOMs to this CycloneDX version before uploading them
+        to Atlas. If the SBOM is already in this version or lower,
+        it will be uploaded as is.
+      type: string
+    - name: supportedSpdxVersion
+      default: "2.3"
+      description: >-
+        If the SBOM uses a higher SPDX version, `syft convert` in the task
+        will convert all SBOMs to this SPDX version before uploading them
+        to Atlas. If the SBOM is already in this version or lower,
+        it will be uploaded as is.
+      type: string
+  workspaces:
+    - name: data
+      description: Directory containing the SBOMs to upload
+  volumes:
+    - name: atlas-secret
+      secret:
+        secretName: $(params.atlasSecretName)
+  stepTemplate:
+    env:
+      - name: sbomsDir
+        value: $(workspaces.data.path)/$(params.sbomsDir)
+      - name: workdir
+        value: /tekton/home
+  steps:
+    - name: gather-sboms
+      image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+      script: |
+        #!/usr/bin/env bash
+        set -o errexit -o pipefail -o nounset
+        
+        skipUploadingSbom=$(params.skipUploadingSbom)
+        if [[ "$skipUploadingSbom" == "true" ]]; then
+          echo "Skipping SBOM uploading - skipUploadingSbom == true"
+          exit 0
+        fi
+
+        version_lesser_equal() {
+          local first
+          first="$(printf "%s\n%s" "$1" "$2" | sort --version-sort | head -n 1)"
+        
+          if [ "$1" = "$first" ]; then
+            echo "SBOM version ($1) is supported (<= $2), will not convert"
+          else
+            echo "SBOM version ($1) is not supported, will convert to $2"
+            printf "%s" "$2" > "${sbom_path}.convert_to_version"
+          fi
+        }
+        
+        echo "Looking for CycloneDX or SPDX SBOMs in $sbomsDir"
+        
+        find "$sbomsDir" -type f | while read -r filepath; do
+          # Skip non-JSON files
+          file_relpath=$(realpath "$filepath" --relative-base="$sbomsDir")
+          if ! jq empty "$filepath" 2>/dev/null; then
+            echo "$file_relpath: not a valid JSON"
+            continue
+          fi
+        
+          # Extract the format information using jq
+          sbom_format=$(jq -r 'if .bomFormat then .bomFormat else if .spdxVersion then "SPDX" else empty end end' "$filepath")
+        
+          # If sbom_format is empty, it's not a valid SBOM (CycloneDX or SPDX)
+          if [ -z "$sbom_format" ]; then
```

Comment:
This is already covered by the `else` on line 149. So either this can be removed. Or maybe it's more readable to have an explicit check at the top, in which case this could cover the other case as well. Either, having two places where we say `not a valid SBOM` and continue the loop seems suboptimal.